repos/system_upgrade/el7toel8/actors/bootentriesscanner/libraries/bootentriesscanner.py

```diff
@@ -0,0 +1,47 @@
+from leapp.models import BootEntry
+from leapp.libraries.stdlib import run, api
+
+
+CMD_GRUBBY_INFO_ALL = ['grubby', '--info', 'ALL']
+
+
+def remove_quotes_if_used(text):
+    """
+    Removes the first pair of quotes surrounding the given text if any.
+
+    :param text: str
+    :rtype: str
+    :return: The given text with the first pair of quotes removed if present.
+    """
+
+    if len(text) < 2:
+        return text
+
+    if text[0] == text[-1] and text[0] in ['\'', '\"']:
+        return text[1:-1]
+    return text
+
+
+def scan_boot_entries():
+    """
+    Scans the available boot entries are informs other actors about the findings
+    via a :class:`BootEntry` message.
+    """
+    grubby_output = run(CMD_GRUBBY_INFO_ALL, split=True)
```

Comment:
I think you want to check that this is not failing by catching the `leapp.libraries.stdlib.CalledProcessError` exception.
Since you didn't specify otherwise, `run`  assumed checked=True which is fine. However you need to handle the exception or, if grubby for whatever reason fails, you will be causing an error in the upgrade and it will exit after finishing the current phase.

Reply:
Just a quick question (so I know how to approach similar matters in the future):
- on `x86` and others - failing to call `grubby`, and therefore, failing to retrieve a list of boot entries is not that big of a problem, as there are no problems known with rescue entries etc. My proposed solution would be to log an error (and probably inform the user via report too) and move on.
- on `s390x` however, we know that without knowing what boot entries there are, the system may end up unable to boot. I suppose, that we definitely want to inform the user about this via a report entry. Just to be sure - should we also preemptively inhibit the upgrade? (I would not say so.)

The lesson I would like to take from this as a novice: In a scenario like this (when some error means that we cannot guarantee a safe upgrade process, but also strictly doesn't mean trouble with the IPU) what stand should I take?

Reply:
The error is handled in the latest commit. Thank you.